src/Sulu/Bundle/WebsiteBundle/Controller/SitemapController.php

```diff
@@ -31,19 +32,48 @@ class SitemapController extends WebsiteController
      */
     public function indexAction(Request $request)
     {
-        $dumpDir = $this->getDumpDir($request);
-        if ($this->get('filesystem')->exists($dumpDir . '/sitemap.xml')) {
-            return $this->createBinaryFileResponse($dumpDir . '/sitemap.xml');
+        if (null !== ($response = $this->getDumpedIndex($request))) {
+            return $response;
         }
 
-        $pool = $this->get('sulu_website.sitemap.pool');
-        if (!$pool->needsIndex()) {
-            return $this->sitemapPaginatedAction($request, $pool->getFirstAlias(), 1);
+        $sitemap = $this->get('sulu_website.sitemap.xml_renderer')->renderIndex();
+        if (!$sitemap) {
+            $aliases = array_keys($this->get('sulu_website.sitemap.pool')->getProviders());
+
+            return $this->sitemapPaginatedAction($request, reset($aliases), 1);
```

Comment:
What is if the only sitemap provider has multiple pages? Then `/sitemap.xml` won't show every URL, would it?